tests/common/src/lib.rs

```diff
@@ -270,16 +271,10 @@ impl<'a> JournalHandler for Journal<'a> {
                 );
             }
             DispatchOutcome::InitSuccess { .. } => {}
-            DispatchOutcome::InitFailure {
-                program_id,
-                message_id,
-                reason,
-                ..
-            } => {
-                panic!(
-                    "Init failure (pid: {:?}, mid: {:?}): {:?}",
-                    program_id, message_id, reason
-                );
+            DispatchOutcome::InitFailure { program_id, .. } => {
+                if let Some(prog) = self.context.programs.get_mut(&program_id) {
+                    *prog = None;
+                }
```

Comment:
Maybe to use `Option::take()`?

```suggestion
                self.context.programs.get_mut(&program_id).take();
```

Reply:
discussed